sw/device/lib/sdk/timer/timer_sdk.c

```diff
@@ -68,6 +83,50 @@ uint32_t hw_timer_stop()
 // Initialize the timer
 void timer_init()
 {
-    // Enable MCYCLE counter
-    CSR_CLEAR_BITS(CSR_REG_MCOUNTINHIBIT, 0x1);
+    // Get current Frequency
+    soc_ctrl_t soc_ctrl;
+    soc_ctrl.base_addr = mmio_region_from_addr((uintptr_t)SOC_CTRL_START_ADDRESS);
+    uint32_t freq_hz = soc_ctrl_get_frequency(&soc_ctrl);
+
+    // Initialize the timer
+    rv_timer_init(timer_base, timer_cfg, &timer);
+    rv_timer_approximate_tick_params(freq_hz, freq_hz, &tick_params);
+    rv_timer_set_tick_params(&timer, 0, tick_params);
+    rv_timer_counter_set_enabled(&timer, 0, kRvTimerEnabled);
+}
+
+// Initialize the timer
+void timer_init_us()
+{
+    // Get current Frequency
+    soc_ctrl_t soc_ctrl;
+    soc_ctrl.base_addr = mmio_region_from_addr((uintptr_t)SOC_CTRL_START_ADDRESS);
+    uint32_t freq_hz = soc_ctrl_get_frequency(&soc_ctrl);
+
+    // Initialize the timer
+    rv_timer_init(timer_base, timer_cfg, &timer);
+    rv_timer_approximate_tick_params(freq_hz, FREQ_US, &tick_params);
```

Comment:
FREQ_US is not a good name, let's call it FREQ_1MHz - also, write somewhere in the comments, APP, and documentation that 1ms is then the mimum time step